pkg/bmt/interface.go

```diff
@@ -19,15 +19,15 @@ const (
 type Hash interface {
 	hash.Hash
 
-	// SetSpan sets the length prefix of BMT hash.
-	SetSpan(int64)
+	// SetMetaLength sets the length prefix of BMT hash.
+	SetMetaToLength(int64)
```

Comment:
comment says `SetMetaLength` but method name is `SetMetaToLength`. Also, I would even go as far as renaming it to be `SetMetaInt64`, since the number can be literally anything. And a small naming nitpick - consider `Preamble` or `Header` instead of `Meta`